File: Python/Leetcode 129.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from typing import Optional


class Solution:
    def Helper(self,root: Optional[TreeNode], temp_sum : int) -> int:
        
        if (root.left == None and root.right == None):
            return temp_sum*10 + root.val

        total_sum = 0
        
        if (root.left != None):
            total_sum += self.Helper(root.left,temp_sum*10 + root.val)
        
        if (root.right != None):
            total_sum += self.Helper(root.right,temp_sum*10 + root.val)
        
        return total_sum


    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.Helper(root,0)
        

File: C++/Leetcode 129.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};



class Solution {
    int Helper(TreeNode* root,string temp)
    {
        if (root->left == NULL && root -> right == NULL)
        {
            // Leaf Node
            return stoi(temp + to_string(root -> val));
        }

        int total_value = 0;

        if (root -> left != NULL)
        {
            total_value += Helper(root -> left, temp + to_string(root -> val));
        }

        if (root -> right != NULL)
        {
            total_value += Helper(root -> right, temp + to_string(root -> val));
        }
        return total_value;
    }
public:
    int sumNumbers(TreeNode* root) {
        return Helper(root,"");
    }
};